### ISSUE TITLE
Consuming message coming in truncated

### ISSUE THREAD

**Issue by yervandakopyan** [php-amqplib/php-amqplib#579]
Im publishing a large json data set (not more than 2MB) however when my consumer picks up the message it is truncated.  Below is my consumer code. Anyone know what is going on or how to fix it. I have tried the latest version and same issue exists

```
<?php
include(__DIR__ . '/config.php');
use PhpAmqpLib\Connection\AMQPStreamConnection;
define('AMQP_WITHOUT_SIGNALS', true);
$exchange = 'exchange';
$queue = 'queue';
$consumerTag = 'consumer' . rand();
$connection = new AMQPStreamConnection(HOST, PORT, USER, PASS, VHOST, false, 'AMQPLAIN', null, 'en_US', 25, 98, null, true, 48);
$channel = $connection->channel();
$channel->setBodySizeLimit(0);
$conn = new ConnectModel();

$channel->queue_declare($queue, false, true, false, false);

$channel->exchange_declare($exchange, 'direct', false, true, false);
$channel->queue_bind($queue, $exchange);


function process_message($message)
{
    syslog(LOG_INFO, 'GOT MESSAGE ' . json_encode($message));
    
    //do something with message here

    $message->delivery_info['channel']->basic_ack($message->delivery_info['delivery_tag']);
    // Send a message with the string "quit" to cancel the consumer.
    if ($message->body === 'quit') {
        $message->delivery_info['channel']->basic_cancel($message->delivery_info['consumer_tag']);
    }
}

$prefetchSize = null; // msg size in bytes, must be null else error
$prefetchCount = 1;
$applyPerChannel = null; // can be false or null, otherwise error
$channel->basic_qos($prefetchSize, $prefetchCount, $applyPerChannel);
$channel->basic_consume($queue, $consumerTag, false, false, false, false, 'process_message');

function shutdown($channel, $connection)
{
        $channel->close();
        $connection->close();
}
register_shutdown_function('shutdown', $channel, $connection);
// Loop as long as the channel has callbacks registered
while (count($channel->callbacks)) {
        $read = array($connection->getSocket()); // add here other sockets that you need to attend
        $write = null;
        $except = null;
        if (false === ($changeStreamsCount = stream_select($read, $write, $except, 60))) {
                /* Error handling */
                echo "Error in stream select!";
        } elseif ($changeStreamsCount > 0) {
                $channel->wait();
        }
}
```

**Comment by lukebakken** [comment]
Are you certain that the entire message is published correctly? Please share your publisher code as well as a typical message.

**Comment by yervandakopyan** [comment]
Yes im certian - i have tried publishing via the library, via a CURL call and i have even published the message through the interface

I have tried the same message cut down to a shorter length manually before publishing and i see the entire message

**Comment by lukebakken** [comment]
This library is maintained by volunteers as well as RabbitMQ core team members when free time permits. Anything you can do to speed up diagnosis of an issue will make it much more likely that your issue will get attention. In this case, it's stuff like this:

* What broker are you using, what version, and what operating system
* If RabbitMQ , what version and what Erlang version
* In this case, sharing your publisher code as well as a typical message

**Comment by yervandakopyan** [comment]
OS: CentOS release 6.9 (Final) (2.6.32-696.el6.x86_64 #1 SMP
RabbitMQ Server: 3.7.7-1.el6.noarch
Erlang Version: 20.1.7-1.el6.x86_64

**Comment by makasim** [comment]
It is not a bug but a "feature" https://github.com/php-amqplib/php-amqplib/blob/08a81740a76af14eb1fb9de09ee3680e6ab82c91/PhpAmqpLib/Channel/AbstractChannel.php#L299

**Comment by lukebakken** [comment]
@makasim that's only if that value is set, which it doesn't appear to be in @yervandakopyan 's case.

**Comment by yervandakopyan** [comment]
anyone have any idea how to resolve this?

**Comment by lukebakken** [comment]
@yervandakopyan - I realize this has taken a while but please, read what I wrote earlier:

> This library is maintained by volunteers as well as RabbitMQ core team members when free time permits. 

I am able to spend some time on this issue this week and will try to reproduce what you report. Thanks for your patience.

**Comment by lukebakken** [comment]
```
$ php --version
PHP 7.2.10 (cli) (built: Sep 11 2018 08:51:16) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
```

I used this script - [repro.php.txt](https://github.com/php-amqplib/php-amqplib/files/2458276/repro.php.txt)

I published the contents of `message.txt` that you provided using the management UI, and got the following result:

[result.txt](https://github.com/php-amqplib/php-amqplib/files/2458273/result.txt)

Note that `message.txt` is 7024 characters and the entire message was received. I am running RabbitMQ 3.7.8 / Erlang 21.1 on Arch Linux and ran the test on the same workstation. I'm not sure what's up in your environment but I suggest doing a packet trace at this point.
